esp-hal/src/i2c/master/mod.rs

```diff
@@ -106,13 +106,54 @@ pub enum Error {
     ZeroLengthInvalid,
 }
 
+/// I2C no acknowledge error reason.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+#[non_exhaustive]
```

Comment:
Why is this `non_exhaustive`, isn't this enum complete? What else is there for the device to not acknowledge?

(Using both `Unknown` and `non_exhaustive` seems wrong to me at a glance)

Reply:
In future we might want to provide more information with the error (e.g. at which byte count the nack was detected) - adding such a variant would be a breaking change then. I imagine user-code will do the same when matching `Unknown` or `_` so I don't think it's a big deal 

I agree for now we wouldn't even need `Address` and `Data` for now, then